crates/common/src/types/mode.rs

```diff
@@ -34,14 +35,19 @@ impl Display for Mode {
 
 impl Mode {
     /// Return all of the available mode combinations.
-    pub fn all() -> impl Iterator<Item = Mode> {
-        ModePipeline::test_cases().flat_map(|pipeline| {
-            ModeOptimizerSetting::test_cases().map(move |optimize_setting| Mode {
-                pipeline,
-                optimize_setting,
-                version: None,
-            })
-        })
+    pub fn all() -> impl Iterator<Item = &'static Mode> {
+        static ALL_MODES: LazyLock<Vec<Mode>> = LazyLock::new(|| {
```

Comment:
This static will be shared across every all_modes instance in the app wouldn't it, so wouldn't work as expected

Reply:
Ah yeah it's all good; when I glanced at this on my phone I had in mind another function!

Personally I wouldn't use a locking static in this way without benchmarking because the iterating should be super cheap anyway but shouldn't be any issue :)